src/utils/parsers/ParserAbstract.ts

```diff
@@ -0,0 +1,14 @@
+import { getConfig } from "../../config";
+
+type codeCompletion = (text : string, filename : string, huggingface_token? : string) => Promise<string>;
```

Comment:
Since this is an abstract class, dont get too specific. the token is too far into specific functionality. implementation of another api without the use of a huggingface token would become weird. maybe just keep it at token?

Reply:
The token is fetched from the config file under the name "huggingface_token", is there a way to provide dynamic token support?

Reply:
There arises a new problem: Dynamic settings fields aren't available for vscode extensions, What should I do? Since each parser needs a unique token, How do I take these in?

Reply:
VSCode allows extensions to add dynamic settings. You can see some examples we have for Captain Stack here https://github.com/hieunc229/copilot-clone/blob/master/package.json#L41

Is that what you're looking for?